### PR TITLE
[Codegen] refactor codegen for poinwise

### PR DESCRIPTION
**Pull request by iclementine**
1. support 0d task space;
2. support specifying output dtypes;
3. support non-tensor arguments to the scalar function.

and remove pointwise_static.py

### CODE REVIEW

File: src/flag_gems/utils/pointwise_dynamic.py [hunk 1]
```diff
@@ -1,172 +1,346 @@
...
+                    code.writeline(f"out{num_output_tensor_index} = torch.empty(shape, dtype=in0.dtype, device=in0.device)")
+                else:
+                    code.writeline(f"out{num_output_tensor_index} = torch.empty(shape, dtype={type_name(op_desc.dtypes[i])}, device=in0.device)")
+                num_output_tensor_index += 1
+
+        # call destination_passing_func
+        call_str = f"{output_ref_for_wrapper(op_desc)} = {destination_passing_func_name}({parameter_ref_for_wrapper(op_desc, include_outputs=True)})"
+        code.writeline(call_str)
+
+        return_str = f"return {output_ref_for_wrapper(op_desc)}"
```
Comment (StrongSpoon) [inline src/flag_gems/utils/pointwise_dynamic.py]: what about storing the return value of output_ref_for_wrapper?

File: src/flag_gems/utils/pointwise_dynamic.py [hunk 2]
```diff
@@ -1,172 +1,346 @@
...
+    # task partitioning, 1d task indexing
+    tile_size = 512
+    num_warps = 4
+    if rank == 0:
+        tile_size = 32
+        num_warps = 1
+
+    with code.indent():
+        # docstring
+        wrapper_docstring = docstring_for_wrapper(op_desc)
```
Comment (StrongSpoon) [inline src/flag_gems/utils/pointwise_dynamic.py]: is it redundant?

File: src/flag_gems/utils/pointwise_dynamic.py [hunk 3]
```diff
@@ -1,172 +1,346 @@
...
+                output_tensor_index += 1
+    return ", ".join(parameters)
+
+def docstring_for_wrapper(op_desc: OPDesc):
+    doc = f'"""Generated destination passing style wrapper function with {op_desc.num_input_tensors()} input tensor(s), {op_desc.num_output_tensors()} output tensor(s), {op_desc.num_non_tensor_args()} non tesnor input(s)."""'
+    return doc
+
+
+def generate_imports(code: IndentedBuffer) -> IndentedBuffer:
+    code.writeline("import math")
```
Comment (StrongSpoon) [inline src/flag_gems/utils/pointwise_dynamic.py:230]: what is "import math" used for?

Comment (iclementine) [reply src/flag_gems/utils/pointwise_dynamic.py:230]: Triton kernel code can contain invocations of functions and use of variables defined in `math`. So it is included for these uses.

File: src/flag_gems/utils/pointwise_dynamic.py (hunk 1 above)
Comment (iclementine) [reply src/flag_gems/utils/pointwise_dynamic.py]: How to sort the return value? 
I return them by the order they are defined in the scalar function.

File: src/flag_gems/utils/pointwise_dynamic.py (hunk 2 above)
Comment (iclementine) [reply src/flag_gems/utils/pointwise_dynamic.py]: The 2 wrapper both have a docstring.

File: src/flag_gems/utils/pointwise_dynamic.py (hunk 1 above)
Comment (StrongSpoon) [reply src/flag_gems/utils/pointwise_dynamic.py]: aren't they of the same order?

File: src/flag_gems/utils/pointwise_dynamic.py (hunk 2 above)
Comment (StrongSpoon) [reply src/flag_gems/utils/pointwise_dynamic.py]: yes but the contents are the same

File: src/flag_gems/utils/pointwise_dynamic.py (hunk 3 above)
Comment (StrongSpoon) [reply src/flag_gems/utils/pointwise_dynamic.py:230]: get

File: src/flag_gems/utils/pointwise_dynamic.py (hunk 2 above)
Comment (iclementine) [reply src/flag_gems/utils/pointwise_dynamic.py]: Done, I have change that.

File: src/flag_gems/utils/pointwise_dynamic.py (hunk 1 above)
Comment (iclementine) [reply src/flag_gems/utils/pointwise_dynamic.py]: Sorry，I misred storing as sorting

**Pull request merged**